src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -23,10 +23,16 @@
  */
 public class TopicManagementService implements Service {
   private final MultiClusterTopicManagementService _multiClusterTopicManagementService;
+  CompletableFuture<Void> _topicPartitionReady = new CompletableFuture<>();
 
   public TopicManagementService(Map<String, Object> props, String serviceName) throws Exception {
     Map<String, Object> serviceProps = createMultiClusterTopicManagementServiceProps(props, serviceName);
     _multiClusterTopicManagementService = new MultiClusterTopicManagementService(serviceProps, serviceName);
+    _topicPartitionReady.complete(null);
```

Comment:
Very, _very_ sneaky. Please refactor.

Reply:
As described above, the asynchronous operations refactoring will be included in a PR